controllers/infrastructure/byomachine_controller.go

```diff
@@ -537,3 +590,52 @@ func (r *ByoMachineReconciler) markHostForCleanup(ctx context.Context, machineSc
 	// Issue the patch for byohost
 	return helper.Patch(ctx, machineScope.ByoHost)
 }
+
+func (r *ByoMachineReconciler) getInstallerConfig(ctx context.Context, byoMachine *infrav1.ByoMachine) (*infrav1.K8sInstallerConfig, error) {
+	installerConfig := &infrav1.K8sInstallerConfig{}
+	installerName := client.ObjectKey{
+		Namespace: byoMachine.Namespace,
+		Name:      byoMachine.Name,
+	}
+	err := r.Client.Get(ctx, installerName, installerConfig)
+	if err != nil {
+		return nil, err
+	}
+	return installerConfig, nil
+}
+
+func (r *ByoMachineReconciler) createInstallationConfig(ctx context.Context, machineScope *byoMachineScope) (*infrav1.K8sInstallerConfig, error) {
+	logger := log.FromContext(ctx).WithValues("cluster", machineScope.Cluster.Name)
+	installerConfig, err := r.getInstallerConfig(ctx, machineScope.ByoMachine)
+	if err != nil && apierrors.IsNotFound(err) {
+		template := &infrav1.K8sInstallerConfigTemplate{}
+		installerTemplateName := client.ObjectKey{
+			Namespace: machineScope.ByoMachine.Spec.InstallerRef.Namespace,
+			Name:      machineScope.ByoMachine.Spec.InstallerRef.Name,
+		}
+		if err = r.Client.Get(ctx, installerTemplateName, template); err != nil {
+			logger.Error(err, "failed to get k8s installer config template")
+			return nil, err
+		}
+
+		installerConfig.ObjectMeta = metav1.ObjectMeta{
+			Name:      machineScope.ByoMachine.Name,
+			Namespace: machineScope.ByoMachine.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(machineScope.ByoMachine, infrav1.GroupVersion.WithKind("ByoMachine")),
+			},
+		}
+		installerConfig.Spec = template.Spec.Template.Spec
+		installerConfig.Annotations = map[string]string{
+			infrav1.K8sVersionAnnotation: *machineScope.Machine.Spec.Version,
```

Comment:
can we use the same way to add k8sversion annotation value as done for byohost: [Ref](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/controllers/infrastructure/byomachine_controller.go#L484)
```sh
infrav1.K8sVersionAnnotation: strings.Split(*machineScope.Machine.Spec.Version, "+")[0]
```